Mathlib/Data/Finset/Lattice.lean

```diff
@@ -124,6 +118,12 @@ theorem le_sup {b : β} (hb : b ∈ s) : f b ≤ s.sup f :=
 theorem le_sup_of_le {b : β} (hb : b ∈ s) (h : a ≤ f b) : a ≤ s.sup f := h.trans <| le_sup hb
 #align finset.le_sup_of_le Finset.le_sup_of_le
 
+theorem sup_union [DecidableEq β] : (s₁ ∪ s₂).sup f = s₁.sup f ⊔ s₂.sup f :=
+  Finset.induction_on s₁
+    (by rw [empty_union, sup_empty, bot_sup_eq])
+    (fun a s _ ih => by rw [insert_union, sup_insert, sup_insert, ih, sup_assoc])
```

Comment:
The mathport output doesn't agree here, its proof is shorter.

Reply:
Found an even shorter proof, at the expense of reordering other lemmas a bit.